src/main/java/com/google/songgame/servlets/RoomServlet.java

```diff
@@ -1,25 +1,62 @@
 package com.google.songgame.servlets;
 
 import com.google.gson.Gson;
+import com.google.gson.JsonParser;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.stream.Collectors;
+import java.util.Collections;
 
 import org.apache.hc.core5.http.ParseException;
 import com.google.gson.Gson; 
-import java.util.ArrayList;
+import java.util.*;
 
 @WebServlet("/room")
 public final class RoomServlet extends HttpServlet {
 
+  private Gson gson;
+
+  @Override
+  public void init() {
+    gson = new Gson();
+  }
+
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // Get user name
-    String name = request.getParameter("user-name");
-    response.setContentType("text/html;");
-    response.getWriter().println("Name: " + name);
+    Map<String, String> userJson = readJSONFromRequest(request);
```

Comment:
Note that this isn't really JSON anymore that we've parsed it into a map. I think a better name would be "userProperties", since we've parsed the JSON into a map with properties about the user (e.g. user name, user ID)